hazelcast/src/main/java/com/hazelcast/map/impl/MapLoadProgressTracker.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl;
+
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.hazelcast.logging.LoggingService;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Class tracking the progress of map entry loading. It logs the
+ * progress in every 10 seconds if there is loading activity, which
+ * triggers logging.
+ *
+ * Note that this class logs the number of records loaded in the last 10
+ * seconds but does not log when the load starts or completes. This is
+ * because the current map loader implementation does not provide
+ * callbacks for these events.
+ */
+public class MapLoadProgressTracker {
```

Comment:
can you please state the behavior of this tracker during load lifecycles? For example, is it created per map? Or when is it reset? For example, if i call `imap#loadAll` multiple times one after another or concurrently what will it print?

Reply:
Added these to the javadoc.